flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/reconciler/BaseReconciler.java

```diff
@@ -66,4 +69,30 @@ private DeleteControl shutdownAndDelete(
     }
 
     protected abstract void shutdown(FlinkDeployment flinkApp, Configuration effectiveConfig);
+
+    protected void updateForSpecReconciliationSuccess(FlinkDeployment flinkApp) {
+        ReconciliationStatus reconciliationStatus = flinkApp.getStatus().getReconciliationStatus();
+        reconciliationStatus.setSuccess(true);
+        reconciliationStatus.setError(null);
+
+        if (reconciliationStatus.getLastReconciledSpec() != null) {
+            int oldSavepointGeneration =
+                    reconciliationStatus.getLastReconciledSpec().getJob().getSavepointGeneration();
+            FlinkDeploymentSpec cloned = CloneUtils.clone(flinkApp.getSpec());
+            cloned.getJob().setSavepointGeneration(oldSavepointGeneration);
+            reconciliationStatus.setLastReconciledSpec(cloned);
+        } else {
+            reconciliationStatus.setLastReconciledSpec(flinkApp.getSpec());
```

Comment:
We could clone/copy `flinkApp.getSpec()` before set it in both branches. They should both be copies for consistency

Reply:
done